### PR TITLE
Adding support for Mesos 0.22.0

### PR DESCRIPTION
**Pull request by rayrod2030**
This PR adds support for Mesos 0.22.0 with the following:

* Updated default Mesos version to 0.21.1.
* Updated available flags in attributes/mesos_options.rb.
* Added new serverspec tests for 0.22.0.
* Added logic for new Mesos binary location which changed from /usr/local/sbin to /usr/sbin for 0.21.1+.
* Using chef provisioner 11.12.4 for a test suite to expand test coverage.
* Removed support for Mesos 0.19.x.
* Updated docs.

### CODE REVIEW

**Comment by rayrod2030** [comment]
Hey @rveznaver I would appreciate your feedback with this update whenever you have a minute.

File: README.md [hunk 1]
```diff
@@ -30,14 +30,14 @@ Supported Mesos Versions
...
+* 0.22.0
 * 0.21.1
 * 0.21.0
 * 0.20.1
 * 0.20.0
-* 0.19.1
 
 We intend to support at most the three latest versions of Apache Mesos including
-all patch releases for supported major versions. Ex. 0.20.0 + 0.20.1 etc..
+all patch releases for supported major versions. Ex. 0.21.0 + 0.21.1 etc..
```
Comment (ankurcha) [inline README.md]: The usage section should also be updated. zookeeper* attributes are no longer valid i believe.

**Comment by ankurcha** [comment]
Another thing to note: In 0.22.0 (packages from mesosphere) the binaries are located under `/usr/sbin` not `/usr/local/sbin`

File: README.md (hunk 1 above)
Comment (rayrod2030) [inline README.md]: Fixed.  Thanks for catching that.

**Comment by rveznaver** [comment]
Hi @rayrod2030, sorry for the late reply, I was without proper internet access for the past couple of days. Looks good to me, the only comment I have is a syntax sugar one (I think a "case" statement would look prettier for the binary location than an "if" one :)).

**Pull request merged**